javalib/src/main/scala/java/util/LinkedList.scala

```diff
@@ -0,0 +1,453 @@
+package java.util
+
+import scala.annotation.tailrec
+
+class LinkedList[E]() extends AbstractSequentialList[E]
+    with List[E] with Deque[E] with Cloneable with Serializable {
+
+  def this(c: Collection[_ <: E]) = {
+    this()
+    addAll(c)
+  }
+
+  import LinkedList._
+
+  var head: Node[Box[E]] = Node()
+  var last: Node[Box[E]] = Node()
+
+  var _size = 0
+
+  def getFirst(): E =
+    if (_size <= 0) throw new NoSuchElementException()
+    else peekFirst()
+
+  def getLast(): E =
+    if (_size <= 0) throw new NoSuchElementException()
+    else peekLast()
+
+  def removeFirst(): E = {
+    val ret = head.value
+
+    if (head.next eq null) {
+      head.value = null
+    } else {
+      val oldHead = head
+      head = oldHead.next
+      head.prev = null
+
+      if (_size == 2) {
+        head.next = null
+        last = Node[Box[E]]()
+      }
+    }
+
+    if (ret eq null) null.asInstanceOf[E]
+    else {
+      _size -= 1
+      ret.inner
+    }
+  }
+
+  def removeLast(): E = {
+    val ret = last.value
+
+    if (_size < 2) {
+      removeFirst()
+    } else {
+      val oldLast = last
+      last = oldLast.prev
+      last.next = null
+
+      if (size == 2) {
+        head = last
+        last = Node[Box[E]]()
+      }
+    }
+
+    if (ret eq null) null.asInstanceOf[E]
+    else {
+      _size -= 1
+      ret.inner
+    }
+  }
+
+  def addFirst(e: E): Unit = {
+    val boxed = Box(e)
+
+    val oldHead =
+      if (head.value ne null) head
+      else null
+
+    head = Node(boxed, next = oldHead)
+
+    _size += 1
+
+    if (oldHead ne null) {
+      oldHead.prev = head
+
+      if ((last.value eq null) && (oldHead.next eq null))
+        last = oldHead
+    }
+  }
+
+  def addLast(e: E): Unit = {
+    val boxed = Box(e)
+
+    if (head.value eq null) addFirst(e)
+    else {
+      val oldLast =
+        if (last.value ne null) last
+        else if (head.value ne null) head
+        else null
+
+      last = Node(boxed, prev = oldLast)
+
+      _size += 1
+
+      if (oldLast ne null)
+        oldLast.next = last
+    }
+  }
+
+  override def contains(o: Any): Boolean = {
+    val boxed = Box(o.asInstanceOf[E])
+    var found = false
+    val iter = iterator()
+    while (iter.hasNext() && !found) {
+      val next = iter.next()
+      if (Box(next) === boxed) {
+        found = true
+      }
+    }
+
+    found
```

Comment:
This can be simplified as `iterator().exists(_ === o)`. This is similar to `AbstractLinkedList.indexOf` at https://github.com/scala-js/scala-js/blob/master/javalib/src/main/scala/java/util/AbstractList.scala#L25-L26